### PR TITLE
bugfix: Prevent parent ListView widgets from snatching scrolling gestures

### PR DESCRIPTION
**Pull request by Binabh**
This is basically copy of https://github.com/fleaflet/flutter_map/pull/717 (since the PR creator does not seem intrested in implementing the commented option and PR is now stale). I saw that the PR is not  being merged because there is no option to developer if they want previous behavior. I have tried to add a bool in mapoptions to determine the type of behavior in the map.

Let me know if anything else is required. I want this feature to be implemented.
Thank You!

### CODE REVIEW

**Comment by rknell** [comment]
I've just tested this out and it works as advertised without any noticeable issues.

However the name of the flag wasn't very intuitive and its implemented as being on by default so it would be a breaking change.

I would suggest something like "captureGesturesInScrollview: true" and have it off by default. But im not a package maintainer so just my 2c.

Otherwise great work, I was just about to go about implementing it myself for a project due out tomorrow.

**Comment by Binabh** [comment]
> However the name of the flag wasn't very intuitive and its implemented as being on by default so it would be a breaking change.
> 
> I would suggest something like "captureGesturesInScrollview: true" and have it off by default. But im not a package maintainer so just my 2c.

Thank You for your comment. I think It is not a big issue to change that. I can do it. I will wait for review from maintainers and make all the changes along with this if they also suggest any.

**Comment by rknell** [comment]
Yeah absolutely! Their opinion matters, I really just wanted to let them know its working for me in the wild which gives them some confidence to merge the pr in any form.

**Comment by kengu** [comment]
@Binabh Can you apply formatting and push the changes? Just run `dart format .` from repo-root.

**Comment by Binabh** [comment]
@kengu done. let me know if anything else is needs to be modified

File: lib/src/map/flutter_map_state.dart
Comment (kengu) [inline lib/src/map/flutter_map_state.dart:222]: This is just duplicated code, which needs to be moved to a new method that can be called from this builder method. This reduce technical dept and makes the code more maintainable. 

```dart
Widget _buildMap() { ... }

...
     return MapStateInheritedWidget(
        mapState: mapState,
        child: Listener(
            onTap: handleTap,
            onLongPress: handleLongPress,
            onDoubleTap: handleDoubleTap,
            child: options.allowPanningOnScrollingParent  
              ? GestureDetector(
                   onTap: _positionedTapController.onTap,
                   onLongPress: _positionedTapController.onLongPress,
                   onTapDown: _positionedTapController.onTapDown,
                   onTapUp: handleOnTapUp,
                   child: _buildMap(),
                 )
              :  GestureDetector(
                    onScaleStart: handleScaleStart,
                    onScaleUpdate: handleScaleUpdate,
                    onScaleEnd: handleScaleEnd,
                    onTap: _positionedTapController.onTap,
                    onLongPress: _positionedTapController.onLongPress,
                    onTapDown: _positionedTapController.onTapDown,
                    onTapUp: handleOnTapUp,
                    child: _buildMap())),
...
```

**Review by kengu — CHANGES_REQUESTED** [review]
LGTM. See my comment on code-duplication. When this is fixed, I'll approve and merge.

File: lib/src/map/flutter_map_state.dart
Comment (Binabh) [reply lib/src/map/flutter_map_state.dart:222]: sure. I will do this

Comment (Binabh) [reply lib/src/map/flutter_map_state.dart]: @kengu my latest commit should fix the code duplication issue. Please review and let me know if any further changes required.

**Comment by Binabh** [comment]
@kengu fixed the code duplication issue. let me know if anything else required

**Review by kengu — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**
